### PR TITLE
IAF neuron initial implementation

### PR DESCRIPTION
**Pull request by cpehle**
Implements a simple integrate and fire neuron with pseudo-derivative gradient. Addresses #179.

### CODE REVIEW

File: norse/torch/functional/iaf.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    """State of a feed forward IAF neuron
+
+    Parameters:
+        v (torch.Tensor): membrane potential
+    """
+
+    v: torch.Tensor
+
+
+def iaf_step(
```
Comment (Jegp) [inline norse/torch/functional/iaf.py:45]: Documentation? Doesn't have to be a lot of text or equations

File: norse/torch/functional/iaf.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    v_new = (
+        v_new
+        + torch.nn.functional.linear(input_tensor, input_weights)
+        + torch.nn.functional.linear(state.z, recurrent_weights)
+    )
+
+    return z_new, IAFState(z_new, v_new)
+
+
+def iaf_feed_forward_step(
```
Comment (Jegp) [inline norse/torch/functional/iaf.py:66]: A brief docstring would be helpful here as well

File: norse/torch/functional/test/test_iaf.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    iaf_feed_forward_step,
+)
+
+
+def test_iaf_feed_forward_step_batch():
+    x = torch.ones(2, 1)
+    s = IAFFeedForwardState(v=torch.zeros(2, 1))
+
+    z, s = iaf_feed_forward_step(x, s)
+    assert z.shape == (2, 1)
```
Comment (Jegp) [inline norse/torch/functional/test/test_iaf.py:16]: I would recommend adding a one line test to validate that the spikes/state actually change over time. Mostly for regression purposes

File: norse/torch/functional/test/test_iaf.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+def test_iaf_step_batch():
+    x = torch.ones(2, 10)
+    s = IAFState(z=torch.zeros(2, 5), v=torch.zeros(2, 5))
+    p = IAFParameters(v_reset=torch.zeros(2, 5))
+    w_in = torch.randn(5, 10)
+    w_rec = torch.randn(5, 5)
+
+    z, s = iaf_step(x, s, w_in, w_rec, p)
+    assert z.shape == (2, 5)
```
Comment (Jegp) [inline norse/torch/functional/test/test_iaf.py:27]: Same as above

**Review by Jegp — APPROVED** [review]
Nice work. I'm happy to merge as-is, but it would be nice with docstrings. 
Modules would also be useful, but we can implement that later.
